aws-predictions/src/main/java/com/amplifyframework/predictions/aws/request/AWSTranslateRequest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.aws.request;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.predictions.models.LanguageType;
+
+import java.util.Objects;
+
+/**
+ * Simple request instance for text translation operation.
+ */
+public final class AWSTranslateRequest {
+    private final String text;
+    private final LanguageType sourceLanguage;
+    private final LanguageType targetLanguage;
+
+    /**
+     * Constructs an instance of {@link AWSTranslateRequest}
+     * without specifying source and target language overrides.
+     * @param text the text to translate
+     */
+    public AWSTranslateRequest(@NonNull String text) {
+        this(text, null, null);
```

Comment:
Amazon Translate does support `auto` in the `SourceLanguageCode` field:

```console
 $ aws translate translate-text \
            --source-language-code "auto" \
            --target-language-code "es" \
            --text "hello world\!" \
            --profile amplify
```
```json
{
    "TranslatedText": "hola mundo!",
    "SourceLanguageCode": "en",
    "TargetLanguageCode": "es"
}
```

Reply:
@jamesonwilliams we DO have an `UNKNOWN` actually haha. I can use that instead I guess.

@jpignata that's actually really cool that it supports that! I think `sourceLanguage` would also be a good thing to display in `TranslateTextResult` for cases like these, no? iOS seems to only return translated text and target language hmm.

We can also totally ignore the fact that Amazon Translate supports "auto" language detection since we currently require the user to provide a default input language anyways in the configuration, so it may be confusing to them like "Wait if I don't specify source language, does that mean it will use my configuration default or automatically detect my input text language?"

Reply:
I think we should have the auto-detection as P0. I can see that it would be easier to always specify an input language in the config. In this case, the language could be "unknown," or "auto," or whatever. The front-end code shouldn't actually need any special branching for this case, we can just pass the value through to the backend and get back a reasonable result as though "auto" were a real language.

Actually, the default value in the config file, as created by the CLI, should really be "auto". A user might override it by using one of the overloaded methods that accepts a `sourceLanguage`.

Reply:
I can definitely see the auto language detection feature being useful, but:
1. CLI doesn't provide the AUTO option as source language
2. It can cause confusion that I described above
    - customer sets `"sourceLang": "es"`
    - calls `translateText(inputText, targetLanguage)`
    - does it try to translate from language of input text? or Spanish?
3. js and iOS doesn't currently have that behavior, so we would need an API review

Reply:
IMO this was just a requirements miss. I didn't know about this feature, but it feels like a P0 for a text translation service. Google Translate certainly has this feature. (1) is a blocker, (3) is something we should schedule and be aware of, but not block on. For (2), if you have a `"sourceLang"` config, I think it needs to be respected for that version of `translateText(inputText, targetLanguage)`, which doesn't specify one itself. If `"sourceLang"` is `"auto"`, then the language is derived from the `inputText`. If source language is `"es"`, then `inputText` is considered as Spanish.

Reply:
Addressed `null` -> `LanguageType.UNKNOWN`
Will have a separate discussion on adding auto language detection support. [Doc](https://quip-amazon.com/143JAhpzu5TT/Native-Predictions-translate-auto-detect-input-language)